org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinDefaultArgumentsFilterTest.java

```diff
@@ -219,4 +219,113 @@ public void should_filter_methods_with_parameters_that_consume_two_slots() {
 		assertIgnored(new Range(m.instructions.get(3), m.instructions.get(3)));
 	}
 
+	/**
+	 * <pre>
+	 * class C(
+	 *   p1: Int,
+	 *   ...
+	 *   p31: Int,
+	 *   p32: Int = 42,
+	 *   p33: Int,
+	 * )
+	 * </pre>
+	 *
+	 * This constructor will have 2 additional parameters containing the mask.
+	 */
+	@Test
+	public void should_filter_methods_with_more_than_32_parameters() {
+		final StringBuilder paramTypes = new StringBuilder();
+		for (int i = 1; i <= 33; i++) {
+			paramTypes.append("I");
+		}
+		final MethodNode m = new MethodNode(InstrSupport.ASM_API_VERSION,
+				Opcodes.ACC_SYNTHETIC, "<init>",
+				"(" + paramTypes
+						+ "IILkotlin/jvm/internal/DefaultConstructorMarker;)V",
+				null, null);
+		context.classAnnotations
+				.add(KotlinGeneratedFilter.KOTLIN_METADATA_DESC);
+
+		m.visitVarInsn(Opcodes.ILOAD, 34);
+		m.visitLdcInsn(1 << 31);
```

Comment:
```suggestion
		m.visitLdcInsn(Integer.valueOf(1 << 31));
```

Avoid autoboxing warning.